### ISSUE TITLE
Celery, nuclei and simple yet stupid nuclei templates (ones that like to match OR statements)

### ISSUE THREAD

**Issue by m00tiny** [yogeshojha/rengine#701]
Lots of scans hanging for seemingly indefinite amounts of time.. particularly with celery restarting or starting the same checks that already take a little while for nuclei to process alone (ones with matching: or at the top of template).. 

### Issue Summary

nuclei scans hanging and celery is making matters worse - 8GB of RAM, 4 vCPUs - ubuntu 22.04 


### Steps to Reproduce

1. Install with script on Ubuntu 22.04
2. Try adding wordpress.com as a target
3. Setup a scan engine that looks for both endpoints and vulnerability scans
4. Watch what happens when it lands on one of these templates that "pretty much always is going to flag a bunch of stuff"
5. misconfiguration/http-missing-security-headers.yaml <----- there's a good example and why not wordpress.com?
6. docker exec -it CONTAINER /bin/sh && ps aux && send sigterm to recover

Maybe timeout? nuclei is supposed to realize it has hung, but the thing is, I can't reproduce this with nuclei alone.. thinking celery isn't helping here..

* I have confirmed that this issue can be reproduced as described on a latest version/pull of reNgine: (yes)


### Technical details
     
Ubuntu 22.04 - The $24 basic shared droplet on digital ocean    
Ubuntu 22.04 - QEMU/KVM running on Manjaro (8GB RAM, 4 vCPUs, eehh.. pretty default much hypervisor defaults.. let me know if you want me to copy paste the XML or what else you might want to know here) here's quick info on the host system (most stuff is shared/passthru)    

```
# mootiny@moofasa: ~                                                                                                                                                                                     (20:35:23)  
ζ inxi -b                                                                                                                                                                                                           
System:
  Host: moofasa Kernel: 5.19.1-3-MANJARO arch: x86_64 bits: 64
    Desktop: KDE Plasma v: 5.24.6 Distro: Manjaro Linux
Machine:
  Type: Laptop System: Micro-Star product: Alpha 15 A3DD v: REV:1.0
    serial: <superuser required>
  Mobo: Micro-Star model: MS-16U6 v: REV:1.0 serial: <superuser required>
    UEFI: American Megatrends v: E16U6AMS.10F date: 03/04/2020
Battery:
  ID-1: BAT1 charge: 38.8 Wh (97.7%) condition: 39.7/53.4 Wh (74.2%)
CPU:
  Info: quad core AMD Ryzen 7 3750H with Radeon Vega Mobile Gfx [MT MCP]
    speed (MHz): avg: 1756 min/max: 1400/2300
Graphics:
  Device-1: AMD Navi 14 [Radeon RX 5500/5500M / Pro 5500M] driver: amdgpu
    v: kernel
  Device-2: AMD Picasso/Raven 2 [Radeon Vega Series / Radeon Mobile Series]
    driver: amdgpu v: kernel
  Device-3: Chicony HD Webcam type: USB driver: uvcvideo
  Display: x11 server: X.Org v: 21.1.4 with: Xwayland v: 22.1.3 driver: X:
    loaded: amdgpu unloaded: modesetting,radeon gpu: amdgpu
    resolution: 1920x1080
  OpenGL: renderer: AMD Radeon Vega 10 Graphics (raven LLVM 14.0.6 DRM 3.47
    5.19.1-3-MANJARO) v: 4.6 Mesa 22.1.6
Network:
  Device-1: Realtek RTL8822CE 802.11ac PCIe Wireless Network Adapter
    driver: rtw_8822ce
  Device-2: Realtek driver: r8169
Drives:
  Local Storage: total: 476.94 GiB used: 300.05 GiB (62.9%)
Info:
  Processes: 387 Uptime: 1d 2h 43m Memory: 60.75 GiB used: 7.66 GiB (12.6%)
  Shell: Zsh inxi: 3.3.21
```

**Comment by ocervell** [comment]
If you are running a scan on all subdomains of `wordpress.com` (probably thousands), it is going to take forever... 

What I do: a simple subdomain-only scan on the main domain first, then individually run an engine containing only `vulnerability_scan` on a subdomain that you are interested in.

This is not really a reNgine issue though (try with `nuclei` alone you'll get the same behavior), although we might want to stop the user from scanning too many things.

**Comment by m00tiny** [comment]
> 

I wasn't able to reproduce this with nuclei alone, which I state under steps to reproduce. Even if I launched several nuclei at once against a large number of targets (though, I generally tested that on my laptop which has completely different specifications *for example, my laptop has 64GB of RAM, not 8GB*). Regardless, I've since found a lot of things I was doing wrong. I think this issue can be closed. I never nailed down one exactly was causing these infinitely long 504 (talking over 24 hours). But, with my poking around, I've since rebuilt that droplet with more knowledge and experience going into it's configuration. So, all I can add from that previous configuration is that it's definitely *not* celery. If anything, as far as I can tell it's inside redis somewhere, but that figures, because I don't know shit about redis. And, I'd definitely put money on your hunch around the queue. In this case, definitely not restricted to subdomains or in my case endpoints/urls.

I'm working on some ideas for continuous scanning, so when I get back into it, I'll make a different thread (if necessary, might be able to just send a pull request instead).
